web/packages/teleterm/src/services/fileStorage/fileStorage.ts

```diff
@@ -44,12 +46,25 @@ export interface FileStorage {
   getFileLoadingError(): Error | undefined;
 }
 
-export async function createFileStorage(opts: {
+/**
+ * createFileStorage reads and parses existing JSON structure from filePath or creates a new file
+ * under filePath with an empty object if the file is missing.
+ *
+ * createFileStorage itself uses blocking filesystem APIs but the functions of the returned
+ * FileStorage interface, such as write and replace, are async.
+ */
+// createFileStorage needs to be kept sync so that initialization of the app in main.ts can be sync.
+// createFileStorage is called only during initialization, so blocking the main process during that
+// time is acceptable.
+//
+// However, functions such as write or replace returned by createFileStorage need to be async as
+// those are called after initialization.
+export function createFileStorage(opts: {
```

Comment:
The two comment styles here struck me as odd but now I understand that the comments under the jsdoc style comments are actually omitted in the documentation. neat trick

Reply:
Yeah, I didn't want the comment about the function internals pollute the JSDoc comment. I could've moved the `//` comment to be inside the function, but since it's about `createFileStorage` in general, I decided to keep it there.